src/dotnet/commands/dotnet-projectmodel-server/ProjectManager.cs

```diff
@@ -200,8 +200,10 @@ private bool ProcessMessage()
                     _configure.Value = message.Payload.GetValue("Configuration");
                     break;
                 case MessageTypes.RefreshDependencies:
+                    _refreshDependencies.Value = false;
+                    break;
                 case MessageTypes.RestoreComplete:
-                    _refreshDependencies.Value = 0;
+                    _refreshDependencies.Value = message.Payload.HasValues ? message.Payload.Value<bool>("Reset") : true;
```

Comment:
Not all the conditions can be met here: look up three lines: https://github.com/dotnet/cli/pull/3446/files#diff-c51b65371dd184ab649c2ab965a544eaR203

```
case MessageTypes.RefreshDependencies:
    _refreshDependencies.Value = false;
    break;
```
The value is set to `false`. If the default value in this case is `false` then above case will reset to `true`.